docs/src/howToGetStarted.md

```diff
@@ -38,105 +281,267 @@ for f in dataFiles
+statistics. GigaSOM exports the [`distributed_mapreduce`](@ref) function that
+can be used as a basic building block for running the computations. For
+example, you can efficiently compute a distributed mean of all your data using
+this code:
 
 ```julia
-using Distributed
-addprocs(2) # the number of workers can be higher than 2 -- generally use the number of CPUs available for the task
-@everywhere using GigaSOM
+distributed_mapreduce(di, sum, +) / distributed_mapreduce(di, length, +)
 ```
+The second parameter of `distributed_mapreduce` describes the "map" step
+(summing/measuring the large matrix), the third is the "reduce" step, which
+takes two intermediate results from 2 workers and reduces it to one.
 
-We will use only the lineage markers (cell surface) for the training of the SOM map
-and extract the expression data:
+Column-wise mean (if you don't want to use `dstat`) is slightly more useful
+(and a bit more complicated):
+
+```julia
+distributed_mapreduce(di, d -> mapslices(sum, d, dims=1), +) ./ distributed_mapreduce(di, x->size(x,1), +)
+```
 
+Finally, for distributed computation of per-cluster mean, the clustering
+information needs to be distributed as well (Fortunately, that is easy, because
+the distributed `mapToGigaSOM` does exactly that).
 
 ```julia
-cc = map(Symbol, lineageMarkers)
-dfSom = daf.fcstable[:,cc]
+mapping = mapToGigaSOM(som, di)
+
+# Optionally, we can transform the mapping to metaclustering with 10 clusters, as above
+metaClusters = cutree(k=10, .....)
+distributed_transform(mapping, m -> metaClusters[m])
+
+# Run the distributed computation (mapreduce can accept multiple datasets as well)
+# `bucketmap` applies a function over buckets, effectively running the mean over clusters,
+# `catmapbuckets` nicely combines the results into an array
+sumscounts = distributed_mapreduce([di, mapping],
+    (d, mapping) -> catmapbuckets(
+        (_,clData) -> (sum(clData), length(clData)),
+	d, 10, mapping),
+    (a,b) -> (((as,al),(bs,bl)) -> ((as+bs), (al+bl))).(a,b))
+```
+
+```
+10×4 Array{Tuple{Float64,Int64},2}:
+ (5949.71, 1228)  (-21.9789, 1228)  (12231.3, 1228)  (12303.1, 1228)
+ (6379.98, 1246)  (12464.3, 1246)   (12427.9, 1246)  (12479.8, 1246)
+ (6513.41, 1294)  (12968.8, 1294)   (12960.7, 1294)  (-28.1922, 1294)
+ (6312.37, 1236)  (-26.7392, 1236)  (6.74384, 1236)  (12401.7, 1236)
+ (6395.73, 1285)  (12867.7, 1285)   (-52.653, 1285)  (-26.9795, 1285)
+ (6229.72, 622)   (10.7578, 622)    (6200.1, 622)    (0.882128, 622)
+ (6141.97, 612)   (6078.56, 612)    (45.9878, 612)   (6079.3, 612)
+ (51.3709, 616)   (23.4306, 616)    (6117.53, 616)   (1.15342, 616)
+ (6177.16, 1207)  (-50.4624, 1207)  (48.8023, 1207)  (-5.549, 1207)
+ (8.56597, 654)   (6536.1, 654)     (-29.2208, 654)  (6539.94, 654)
 ```
 
-Initialize the SOM grid by size and expression values:
+This can be aggregated by distributing the `/` to tuples:
+```julia
+clusterMeans = (x->(/)(x...)).(sumcounts)
```

Comment:
Not sure this syntax is very handy ... Is there a different way?

Reply:
something like:
```julia
clusterMeans = zeros(size(sumcounts))
for (i,j) in eachindex(sumcounts)   #hopefully this works, otherwise a nested for-cycle
  clusterMeans[i,j] = sumcounts[i,j][1]/sumcounts[i,j][2]
end
```

Reply:
Yeah, I think that would be better. What do you think @oHunewald ?

Reply:
Oh great, I just found that [] somehow magically preserves array dimensions (not sure if that's even documented somewhere -- looks like the array comprehension syntax is in fact a shortcut for broadcasting the inline function...:D ). So anyway this is probably best:
```julia
clusterMeans = [ sum/count for (sum,count) in sumcounts ]
```

Reply:
:+1: